include/exadg/compressible_navier_stokes/postprocessor/postprocessor.cpp

```diff
@@ -101,36 +102,52 @@ PostProcessor<dim, Number>::do_postprocessing(VectorType const & solution,
   /*
    *  write output
    */
-  output_generator.evaluate(solution, additional_fields, time, time_step_number);
+  if(output_generator.time_control.needs_evaluation(time, time_step_number))
+    output_generator.evaluate(solution,
+                              additional_fields,
+                              time,
+                              Utilities::is_unsteady_timestep(time_step_number));
 
   /*
    *  write pointwise output
    */
-  pointwise_output_generator.evaluate(solution, time, time_step_number);
+  if(pointwise_output_generator.time_control.needs_evaluation(time, time_step_number))
+    pointwise_output_generator.evaluate(solution,
+                                        time,
+                                        Utilities::is_unsteady_timestep(time_step_number));
   /*
    *  calculate error
    */
-  error_calculator.evaluate(solution, time, time_step_number);
+  if(error_calculator.time_control.needs_evaluation(time, time_step_number))
+    error_calculator.evaluate(solution, time, Utilities::is_unsteady_timestep(time_step_number));
 
   /*
    *  calculation of lift and drag coefficients
    */
-  lift_and_drag_calculator.evaluate(velocity, pressure, time);
+  if(lift_and_drag_calculator.time_control.needs_evaluation(time, time_step_number))
+    lift_and_drag_calculator.evaluate(velocity, pressure, time);
 
   /*
    *  calculation of pressure difference
    */
-  pressure_difference_calculator.evaluate(pressure, time);
+  if(pressure_difference_calculator.time_control.needs_evaluation(time, time_step_number))
+    pressure_difference_calculator.evaluate(pressure, time);
 
   /*
    *  calculation of kinetic energy
    */
-  kinetic_energy_calculator.evaluate(velocity, time, time_step_number);
+  if(kinetic_energy_calculator.time_control.needs_evaluation(time, time_step_number))
+    kinetic_energy_calculator.evaluate(velocity,
+                                       time,
+                                       Utilities::is_unsteady_timestep(time_step_number));
 
   /*
    *  calculation of kinetic energy spectrum
    */
-  kinetic_energy_spectrum_calculator.evaluate(velocity, time, time_step_number);
+  if(kinetic_energy_spectrum_calculator.time_control.needs_evaluation(time, time_step_number))
+    kinetic_energy_spectrum_calculator.evaluate(velocity,
+                                                time,
+                                                Utilities::is_unsteady_timestep(time_step_number));
```

Comment:
From these lines, I get the impression that the check whether it is a steady or unsteady time step should be done outside the postprocessor, and `TimeControl` should not take care of this steady/unsteady check, but should address unsteady, time-dependent problems only. Again, this should be done in a subsequent PR. I just write this here so that we do not forget about this.